core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/item/SearchForClassItem.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright (c) 2020 THALES GLOBAL SERVICES.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *  
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.ui.search.searchfor.item;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.emf.common.notify.AdapterFactory;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.edit.provider.ComposedAdapterFactory;
+import org.eclipse.swt.graphics.Image;
+import org.polarsys.capella.core.model.handler.provider.CapellaAdapterFactoryProvider;
+import org.polarsys.capella.core.ui.search.searchfor.GetImagesFromEClassUtil;
+
+/**
+ * 
+ * Search item for EClass
+ */
+public class SearchForClassItem implements SearchForItem {
+  private Object obj;
+
+  public SearchForClassItem(Object obj) {
+    this.obj = obj;
+  }
+
+  /**
+   * Check if this search item covers the input object
+   * 
+   * @param eObj
+   */
+  public boolean cover(Object object) {
+    return object instanceof EObject && obj instanceof EClass && ((EObject) object).eClass() == obj;
+  }
+
+  @Override
+  public String getText() {
+    return ((EClass) obj).getName();
+  }
+
+  @Override
+  public Image getImage() {
+    AdapterFactory adapterFactory = CapellaAdapterFactoryProvider.getInstance().getAdapterFactory();
```

Comment:
@yugansan has added the helper in EObjectLaProHelper

Reply:
It's not yet on this branch. We should think about it when this is merged to 1.4.x.